src/main/java/seedu/address/logic/commands/AddProcCommand.java

```diff
@@ -118,6 +121,9 @@ private Client clientWithAddedProcFrom(Client clientToAddProc) throws CommandExc
             updatedProcedureList.add(procedureList.get(i));
         }
         updatedProcedureList.add(procedure);
+        Comparator<Procedure> mapComparator = (Procedure m1, Procedure m2) -> m1.getDate()
+                .compareTo(m2.getDate());
+        Collections.sort(updatedProcedureList, mapComparator);
```

Comment:
Nice use of Collections! :)

Reply:
Thanks! I felt that the usage of a comparator in conjunction with collections will be much more efficient than creating a new sorting method.